storage/storage_src.go

```diff
@@ -453,9 +453,12 @@ func (s *storageImageSource) getSize() (int64, error) {
 		if err != nil {
 			return -1, err
 		}
-		if (layer.TOCDigest == "" && layer.UncompressedDigest == "") || layer.UncompressedSize < 0 {
+		if (layer.TOCDigest == "" && layer.UncompressedDigest == "") || (layer.TOCDigest == "" && layer.UncompressedSize < 0) {
 			return -1, fmt.Errorf("size for layer %q is unknown, failing getSize()", layerID)
 		}
+		if layer.UncompressedSize < 0 {
```

Comment:
If a TOC-identified layer can have `UncompressedSize == -1`, that breaks more assumptions. At least the `LayersByTOCDigest` path in `tryReusingBlobAsPending` is using the `UncompressedSize` value, and the value can’t be “unknown” on that path.

… but, also, note to self: is a _new_ behavior of the `AdditionalLayerStore` code? We did previously assume that the blobDigest-identified ALS layers did have a valid `UncompressedSize`. Was that always broken?

Reply:
> At least the `LayersByTOCDigest` path in `tryReusingBlobAsPending` is using the `UncompressedSize` value, and the value can’t be “unknown” on that path.

That’s actually fine because that is already conditional on `UncompressedDigest != ""`, and that’s false for recent ALS layers.

Reply:
… and I don’t know whether `UncompressedSize` was _always_ returned with the older version of the FUSE server, but that’s not a concern going forward anyway.